src/osd/osd_types.h

```diff
@@ -4387,13 +4379,17 @@ ostream &operator<<(ostream &lhs, const store_volatile_statfs_t &rhs);
 * ObjectStore full statfs information
 */
 struct store_statfs_t : public store_volatile_statfs_t {
```

Comment:
Instead of separating out these types at this level, which is sort of confusing, I would just defined a separate bluestore-specific type that has the int64_t array structure/alignment and copy the ~5 fields explicitly in BlueStore::statfs().

The bluestore type could even be an int64_t array with an enum for indexes or something?  Maybe